crates/biome_js_parser/src/lexer/mod.rs

```diff
@@ -609,17 +609,19 @@ impl<'src> JsLexer<'src> {
         }
     }
 
-    // Read a `\u0000` escape sequence, this expects the current char to be the `u`, it also does not skip over the escape sequence
-    // The pos after this method is the last hex digit
-    fn read_unicode_escape(&mut self, advance: bool) -> Result<char, ()> {
+    /// Reads a `\u0000` escape sequence.
+    ///
+    /// This expects the current char to be the `u`. Afterwards, the current
+    /// char is the last hex digit.
+    ///
+    /// This returns a `u32` since not all escape sequences produce valid
+    /// Unicode characters.
+    fn read_unicode_escape(&mut self) -> Result<u32, ()> {
         self.assert_byte(b'u');
 
-        for idx in 0..4 {
+        for _ in 0..4 {
             match self.next_byte_bounded() {
                 None => {
-                    if !advance {
-                        self.position -= idx + 1;
-                    }
```

Comment:
Simplified this function a little bit by removing the argument that was only used in one place anyway.